tools/tertiary-analysis/scmap/scmap_index_cell.xml

```diff
@@ -5,12 +5,13 @@
     </macros>
     <expand macro="requirements" />
     <command detect_errors="exit_code"><![CDATA[
-        scmap-preprocess-sce.R --input-object "${input_single_cell_experiment}" --output-sce-object "${input_single_cell_experiment}.preprocessed" && scmap-index-cell.R --input-object-file "${input_single_cell_experiment}.preprocessed" --number-chunks '$n_chunks' --number-clusters '$n_clusters' --output-object-file '$output_single_cell_experiment' --random-seed '$random_seed'
+        scmap-preprocess-sce.R --input-object "${input_single_cell_experiment}" --output-sce-object "${input_single_cell_experiment}.preprocessed" && scmap-index-cell.R --input-object-file "${input_single_cell_experiment}.preprocessed" --train-idf '${train_idf}' --number-chunks '$n_chunks' --number-clusters '$n_clusters' --output-object-file '$output_single_cell_experiment' --random-seed '$random_seed'
     ]]></command>
     <inputs>
         <param type="data" name="input_single_cell_experiment" label="SingleCellExperiment object" format="rdata" help="File with serialized SingleCellExperiment object as produced by 'scmap select features'" />
         <param name="n_chunks" type="integer" label="Number of chunks" value='' help="Number of chunks into which the expr matrix is split" />
         <param name="n_clusters" type="integer" label="Number of clusters" value='' help="Number of clusters per group for k-means clustering" />
+        <param type="data" name="train_idf" label="IDF file" format="txt" help="IDF file to extract training dataset name from" />
```

Comment:
Is this IDF as in MAGETAB? Does SCMAP use IDF? Can the format be set to something that Galaxy recognizes? Otherwise we would need to add in the help how the IDF file needs to look here in terms of format.

Reply:
Yes it's simply a way of keeping track of dataset of origin. It's not an original scmap parameter. Doesn't Galaxy recognise txt files?

Reply:
Well, it would seem to me that the tool should work with out it as well, correct? in which case the input should be made optional (and handled accordingly in the command line part). If this is an IDF file, and galaxy supports such format (we would need to check), then it should be preferred against txt to protect users from just providing a txt with just any content inside (which will still be txt, but not the txt that you need here for a correct run). If Galaxy doesn't support idf, then make sure that the help in the Galaxy tool explains very clearly how this input file should be formatted.

Reply:
okay got it, thanks! @pcm32

Reply:
seems like galaxy does not support IDF files explicitly (https://usegalaxy.org/static/formatHelp.html#tab), so I'll go with the second option

Reply:
would it be okay if I provide this link to a user? https://www.ebi.ac.uk/arrayexpress/help/creating_an_idf.html

Reply:
my other question would be, do we need all the complexity of an IDF for this? how many fields are you looking to capture? I'm concerned that asking for an IDF poses a huge burden on the tool usage for the regular user, which is not part of the original tool that we are wrapping. If the IDF is optional, then that would be fine, but if it is not, I would find it troubling. Maybe we can have a chat to discuss this.

Reply:
@pcm32 Hi Pablo, so we decided to use only dataset ID here rather than parse an IDF file. Please see the latest commit